src/read_write/ply.rs

```diff
@@ -525,61 +530,80 @@ impl NodeWriter<Point> for PlyNodeWriter {
 
 impl Drop for PlyNodeWriter {
     fn drop(&mut self) {
-        self.writer.write_all(b"\n").unwrap();
-        if self
-            .writer
-            .seek(SeekFrom::Start(HEADER_START_TO_NUM_VERTICES.len() as u64))
-            .is_ok()
-        {
-            let _res = write!(
-                &mut self.writer,
-                "{:0width$}",
-                self.point_count,
-                width = HEADER_NUM_VERTICES.len()
-            );
+        if self.point_count == 0 {
+            return;
         }
+        self.writer.write_all(b"\n").unwrap();
+        self.writer.flush().unwrap();
+
+        let mut file = self.writer.get_mut().get_mut();
```

Comment:
Could you comment on why this was changed?

Reply:
Since we now have an append mode, we need to update the number of vertices written. It was much easier to directly use the underlying `File` writer (which implements both the `Read` and `Write` traits) for this instead of adding read functionality to the `DataWriter`, which internally holds a `BufWriter`.

Reply:
Hmm, could `PlyNodeWriter::new()` also just open the file (if any) in read mode and initialize `point_count` from the contents?

Reply:
Yes, this would be another alternative, but then we would need two internal variables `point_count_total` and `point_count_this_session` or the like to keep track of the states...

Reply:
Where do we need to distinguish these from just their sum?

Reply:
We currently have `if self.point_count == 0` at a couple of places, these could also be replaced with a boolean variable and checking `if !self.points_written`, but at least there needs to be a way to find out whether we need to create a header... If you think it's nicer this way, I can change it.

Reply:
Is the reason we can't just write the header every time that some code assumes that only empty files can be safely deleted?

Reply:
As discussed offline, I updated the logic to read in the point count.